masu/api/status.py

```diff
@@ -20,42 +20,87 @@
 import logging
 import os
 import platform
+import socket
 import subprocess
 import sys
 
-from flask import jsonify
+from flask import jsonify, request
 
 from masu.api import API_VERSION
+from masu.celery import celery as celery_app
 from masu.config import Config
 from masu.external.date_accessor import DateAccessor
 from masu.util.blueprint import application_route
 
-logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+LOG = logging.getLogger(__name__)
 
 API_V1_ROUTES = {}
 
 
 @application_route('/status/', API_V1_ROUTES, methods=('GET',))
 def get_status():
     """Packages response for class-based view."""
-    status = ApplicationStatus()
+    if 'liveness' in request.args:
+        return jsonify({'alive': True})
+
+    app_status = ApplicationStatus()
     response = {
-        'commit': status.commit,
-        'python_version': status.python_version,
-        'platform_info': status.platform_info,
-        'modules': status.modules,
-        'api_version': status.api_version,
-        'current_datetime': status.current_datetime,
-        'debug': status.debug
+        'celery_status': app_status.celery_status,
+        'commit': app_status.commit,
+        'python_version': app_status.python_version,
+        'platform_info': app_status.platform_info,
+        'modules': app_status.modules,
+        'api_version': app_status.api_version,
+        'current_datetime': app_status.current_datetime,
+        'debug': app_status.debug
     }
     return jsonify(response)
 
 
-# pylint: disable=too-few-public-methods
+# pylint: disable=too-few-public-methods, no-self-use
 class ApplicationStatus():
     """A view that returns status JSON."""
 
-    def _get_commit(self):  # pylint: disable=no-self-use
+    api_version = API_VERSION
+
+    @property
+    def celery_status(self):
+        """Determine the status of our connection to Celery.
+
+        :returns: dict of celery status
+        """
+        state = celery_app.events.State()
+        events = {}
+
+        def announce_worker_event(event):
+            """Announce worker events."""
+            state.event(event)
+            LOG.debug(f'EVENT: {event}')
```

Comment:
It looks like this would only be printed when the status endpoint is hit.  Would this ever spam?  If not, could we make it `LOG.info`?

Reply:
In general, I don't think we'd ever really want to see this output outside of a development environment. So, I guess my only objection to changing it is to ask whether you think we're currently over-using INFO when we should be using DEBUG?

Reply:
Mainly for debugging in production.  If it's just a line that prints out when the status endpoint is hit, I'd say make it `info`.

That gets to the next part of your question.  When to use `debug`.  I've only found `debug` useful when it's going to seriously spam your log and the noise it produces makes it harder to get to the root cause of the problem your facing.  The debug info is only valuable for very fine grain detail of whats going on.  Think of my temporary loop in the data expiration PR where I'm printing out every single line item being deleted.  There have been *many* times where I've been debugging a production problem and became very frustrated when `debug` was preventing me from getting the info I really needed :)

Reply:
That makes sense. So, in this case, I think DEBUG is still appropriate here. 

The contents of the event dict are exactly the same as what is showing up in the status, with a slight change in the structure (we pop `hostname` off the dict to use it as the status dict's key). 

So, it'll be redundant info to what's already getting logged and printed. The only time we'll care about it is during development.